### ISSUE TITLE
Test suite emits uncaught IERS DEBUG message

### ISSUE THREAD

**Issue by aarchiba** [nanograv/PINT#1277]
Normally `pytest`  should capture all warning messages, but (perhaps because of `loguru`) it emits a (colourful) message partway through the test suite. This message also emerges many other unexpected places (for example in a script that does no log configuration).
```
$ make test
pytest -n auto --ff --durations=20
===================================== test session starts ======================================
platform linux -- Python 3.8.0, pytest-7.1.2, pluggy-1.0.0
rootdir: /home/naa280/projects/pint/PINT
plugins: forked-1.3.0, cov-3.0.0, xdist-2.3.0, hypothesis-6.46.2, anyio-3.2.1
gw0 [1462] / gw1 [1462] / gw2 [1462] / gw3 [1462] / gw4 ok / gw5 [1462] / gw6 [1462] / gw7 [1462gw0 [1462] / gw1 [1462] / gw2 [1462] / gw3 [1462] / gw4 [1462] / gw5 [1462] / gw6 [1462] / gw7 [1462]
..............s.....................s................................................... [  6%]
.....................s.....s.....s.s.................................................... [ 12%]
.........................X...................................s.......................... [ 18%]
................................s..........s.........ss.......x.xs..2022-06-09 14:54:23.189 | DEBUG    | pint.erfautils:<module>:71 - Running get_iers_up_to_date() to update IERS B table, and checking for updated leap seconds.
..x.s..Xs..s...X...X [ 24%]
```

**Comment by aarchiba** [comment]
This appears in the tutorials even with the recommended logging setup: https://nanograv-pint.readthedocs.io/en/latest/examples/time_a_pulsar.html

**Comment by dlakaplan** [comment]
I don't know why pytest doesn't capture this message.  But I think the reason it's seen elsewhere is that it's emitted when importing `erfautils`, which is before any user-specific log setup has been done.  So it just uses the default `loguru` setup.   Ways to fix that could include:
* changing the level from `DEBUG` to `TRACE` or something similar so that it's not seen with the default setup
* removing the logging statement altogether (is it actually useful?)
* moving the relevant lines to somewhere that doesn't always run on import but happens later, after logging setup.  It seems that having some functions called every time on import isn't optimal, but maybe there is a reason for that.

**Comment by aarchiba** [comment]
I think if we can move away from code being run on import we should, it's asking for just this kind of trouble.

**Comment by dlakaplan** [comment]
agreed.  I was just looking for where other clock checks/updates are done to see if this can be included with them.

**Comment by aarchiba** [comment]
Observatory clock corrections are loaded when the first clock correction from that observatory is requested. (This is occasionally annoying when one wants to inspect the state of the clock corrections, but observatory clock corrections are moderately large text files most of which are not needed in most PINT runs.)

I believe Astropy's Time object does any downloading/reading-in when it is asked to convert scales, or at least when it is created, not before.

**Comment by aarchiba** [comment]
Closed by #1281
